In [ ]:
1.#check Maximum number of a list of numbers in Python
numbers = [1, 2, 3, 4, 5]
maximum = max(numbers)
print(maximum)

5


The output of this code is 5 because the max() function in Python iterates through the provided list and returns the element with the highest value. In the list [1, 2, 3, 4, 5], the number 5 is the largest, hence it is returned as the maximum

In [ ]:
2.#python Program for factorial of a number
def factorial(n):
  if n == 0:
    return 1
  else:
    return n * factorial(n - 1)

number = 5
result = factorial(number)
print(f"The factorial of {number} is {result}")


The factorial of 5 is 120


The output of this code is "The factorial of 5 is 120" because the factorial of a non-negative integer 'n', denoted by n!, is the product of all positive integers less than or equal to 'n'.

In this case, 5! is calculated as 5 * 4 * 3 * 2 * 1 = 120. The provided Python code recursively calculates this value.




In [ ]:
3.#python program for simple interest.
def simple_interest(principal, rate, time):
  """Calculates simple interest.

  Args:
    principal: The principal amount.
    rate: The interest rate (as a decimal).
    time: The time period (in years).

  Returns:
    The simple interest.
  """
  return (principal * rate * time) / 100

# Example usage
p = 1000
r = 5
t = 2
si = simple_interest(p, r, t)
print("Simple Interest:", si)


Simple Interest: 100.0


The output of this code is 'Simple Interest: 100.0' because the simple interest is calculated using the formula (P * R * T)/100, where P is the principal amount, R is the rate of interest, and T is the time period.

In this specific example, the principal amount is 1000, the rate of interest is 5%, and the time period is 2 years. Substituting these values into the formula, we get (1000 * 5 * 2)/100 = 100. Therefore, the calculated simple interest is 100.0.

In [ ]:
4.#python program for compound interest.
def compound_interest(principal, rate, time):
  """Calculates compound interest.

  Args:
    principal: The principal amount.
    rate: The interest rate (as a decimal).
    time: The time period (in years).

  Returns:
    The compound interest.
  """
  amount = principal * (1 + rate / 100) ** time
  ci = amount - principal
  return ci

# Example usage
p = 1000
r = 5
t = 2
ci = compound_interest(p, r, t)
print("Compound Interest:", ci)

Compound Interest: 102.5


The output of this code is "Compound Interest: 102.5" because the compound interest is calculated using the formula:

A = P(1 + R/100)^T
CI = A - P

where:

A is the final amount,
P is the principal amount,
R is the interest rate, and
T is the time period.

In this example, P = 1000, R = 5, and T = 2. Substituting these values into the formula, we first calculate A:
A = 1000(1 + 5/100)^2 = 1102.5
Then, we calculate the compound interest (CI):
CI = 1102.5 - 1000 = 102.5
Therefore, the calculated compound interest is 102.5.




In [ ]:
5.#python program to check amstrong number.
def is_armstrong_number(num):
  """Checks if a number is an Armstrong number.

  Args:
    num: The number to check.

  Returns:
    True if the number is an Armstrong number, False otherwise.
  """
  order = len(str(num))
  sum = 0
  temp = num
  while temp > 0:
    digit = temp % 10
    sum += digit ** order
    temp //= 10
  return num == sum

# Example usage
number = 153
if is_armstrong_number(number):
  print(number, "is an Armstrong number")
else:
  print(number, "is not an Armstrong number")

153 is an Armstrong number


The output of this code is "153 is an Armstrong number" because an Armstrong number is a number that is equal to the sum of the cubes of its digits.

In this case, 153 is an Armstrong number because:
1^3 + 5^3 + 3^3 = 1 + 125 + 27 = 153
The provided Python code checks this condition by first determining the number of digits in the input number, then iteratively extracting each digit, calculating its cube, and adding it to a sum. Finally, it compares this sum to the original number to determine if it is an Armstrong number.

In [ ]:
6.#python program for program to find area of a circle.
import math

def area_of_circle(radius):
  """Calculates the area of a circle.

  Args:
    radius: The radius of the circle.

  Returns:
    The area of the circle.
  """
  return math.pi * radius ** 2

# Example usage
radius = 5
area = area_of_circle(radius)
print("Area of the circle:", area)

Area of the circle: 78.53981633974483


The output of this code is "Area of the circle: 78.53981633974483" because the area of a circle is calculated using the formula:
Area = π * r^2
where π (pi) is a mathematical constant approximately equal to 3.14159, and r is the radius of the circle.

In this example, the radius is given as 5. Substituting these values into the formula, we get:

Area = 3.14159 * 5^2 = 78.53981633974483
Therefore, the calculated area of the circle is approximately 78.53981633974483.



In [ ]:
7.#python program to print all prime numbers in an interval.
def is_prime(num):
  """checks if a number is prime or not.
  Args:
    num: The number to check.
    Returns:
    True if the number is prime, False otherwise.
 """
  if num <= 1:
    return False
  i = 2
  while i * i <= num:
    if num % i == 0:
      return False
    i += 1
    return True


lower = 10
upper = 20

print("prime numbers between" , lower, "and , upper, are:")

for num in range(lower, upper +1):
  if is_prime(num):
    print(num)



prime numbers between 10 and , upper, are:
11
13
15
17
19


In [ ]:
8.#python program to check wheather a number is prime or not.
def is_prime(num):
  """Checks if a number is prime.

  Args:
    num: The number to check.

  Returns:
    True if the number is prime, False otherwise.
  """
  if num <= 1:
    return False
  if num <= 3:
    return True
  if num % 2 == 0 or num % 3 == 0:
    return False
  i = 5
  while i * i <= num:
    if num % i == 0 or num % (i + 2) == 0:
      return False
    i += 6
  return True

# Example usage
number = 17
if is_prime(number):
  print(number, "is a prime number")
else:
  print(number, "is not a prime number")

17 is a prime number


This is because the code iterates through numbers in
the interval from 'lower' (10) to 'upper' (20) inclusive, and for each number, it calls the 'is_prime' function to determine if it's a prime number.

A prime number is a natural number greater than 1 that is not a product of two smaller natural numbers.

The 'is_prime' function checks this condition by iterating up to the square root of the number, checking if it's divisible by any smaller number. If no divisors are found, it returns True (indicating a prime number), otherwise False.

Therefore, the code correctly identifies and prints the prime numbers within the given interval: 11, 13,15 17, and 19.

The output of this code is "17 is a prime number" because the provided Python code checks if a given number is a prime number.

A prime number is a natural number greater than 1 that has no positive divisors other than 1 and itself. The function is_prime implements an efficient algorithm to determine primality.

In this specific case, the number 17 is tested. It is not divisible by 2, 3, or any number up to its square root. Therefore, the function correctly identifies 17 as a prime number and prints the corresponding message.



In [ ]:
9.#python program for n-th fibonacci number.
def fibonacci(n):
  """Calculates the n-th Fibonacci number.

  Args:
    n: The position of the desired Fibonacci number.

  Returns:
    The n-th Fibonacci number.
  """
  if n <= 0:
    return 0
  elif n == 1:
    return 1
  else:
    return fibonacci(n - 1) + fibonacci(n - 2)

# Example usage
n = 10
result = fibonacci(n)
print("The", n, "-th Fibonacci number is:", result)


The 10 -th Fibonacci number is: 55


The output of this code is "The 10 -th Fibonacci number is: 55" because the Fibonacci sequence is a series of numbers in which each number is the sum of the two preceding numbers.

The sequence starts with 0 and 1. So, the first 12 numbers in the Fibonacci sequence are: 0, 1, 1, 2, 3, 5, 8, 13, 21, 34, 55, 89.

The provided Python code calculates the n-th Fibonacci number recursively. When n is 10, it calculates the 10th Fibonacci number, which is 55.


In [ ]:
10.#python program for how to check if a given number is fibonacci number.
def is_perfect_square(x):
  """Checks if a number is a perfect square.

  Args:
    x: The number to check.

  Returns:
    True if the number is a perfect square, False otherwise.
  """
  s = int(x ** 0.5)
  return s * s == x

def is_fibonacci(num):
  """Checks if a number is a Fibonacci number.

  Args:
    num: The number to check.

  Returns:
    True if the number is a Fibonacci number, False otherwise.
  """
  return is_perfect_square(5 * num * num + 4) or is_perfect_square(5 * num * num - 4)

# Example usage
number = 8
if is_fibonacci(number):
  print(number, "is a Fibonacci number")
else:
  print(number, "is not a Fibonacci number")

8 is a Fibonacci number


The output of this code is "8 is a Fibonacci number" because a number is a Fibonacci number if and only if one or both of (5n^2 + 4) or (5n^2 – 4) is a perfect square.

In this case, for n = 8, (5 * 8^2 - 4) = 316, which is not a perfect square. However, (5 * 8^2 + 4) = 324, which is a perfect square (18^2). Therefore, the number 8 satisfies the condition and is identified as a Fibonacci number.



In [ ]:
11.# python program for n/'th multiple of a number in fibonacci series.
def find_nth_multiple_in_fibonacci(k, n):
  """Finds the n-th multiple of k in the Fibonacci sequence.

  Args:
    k: The number whose multiple to find.
    n: The position of the desired multiple in the Fibonacci sequence.

  Returns:
    The n-th multiple of k in the Fibonacci sequence.
  """
  count = 0
  a = 0
  b = 1
  while count < n:
    c = a + b
    if c % k == 0:
      count += 1
    a = b
    b = c
  return c

# Example usage
k = 3
n = 4
result = find_nth_multiple_in_fibonacci(k, n)
print("The", n, "-th multiple of", k, "in the Fibonacci sequence is:", result)

The 4 -th multiple of 3 in the Fibonacci sequence is: 987


In [ ]:
12.#python program for sum of squares of first n natural numbers.
def sum_of_squares(n):
  """Calculates the sum of squares of the first n natural numbers.

  Args:
    n: The number of natural numbers to consider.

  Returns:
    The sum of squares of the first n natural numbers.
  """
  return (n * (n + 1) * (2 * n + 1)) // 6

# Example usage
n = 5
result = sum_of_squares(n)
print("Sum of squares of first", n, "natural numbers:", result)


Sum of squares of first 5 natural numbers: 55


The output of this code is "The 4 -th multiple of 3 in the Fibonacci sequence is:987" because the code finds the n-th multiple of a given number 'k' in the Fibonacci sequence.

In this case, it searches for the 4th multiple of 3. By iteratively generating Fibonacci numbers, it checks if each number is divisible by 3. When it encounters the 4th number divisible by 3 , it returns this value as the result.

The output of this code is "Sum of squares of first 5 natural numbers: 55" because the sum of squares of the first 'n' natural numbers is given by the formula:

n * (n + 1) * (2n + 1) / 6
In this case, n = 5. Substituting this value into the formula, we get:

5 * (5 + 1) * (2 * 5 + 1) / 6 = 5 * 6 * 11 / 6 = 55
Therefore, the sum of squares of the first 5 natural numbers is 55.



In [ ]:
13.#python program for cube sum of first n natural numbers.
def sum_of_cubes(n):
  """Calculates the sum of cubes of the first n natural numbers.

  Args:
    n: The number of natural numbers to consider.

  Returns:
    The sum of cubes of the first n natural numbers.
  """
  return (n * (n + 1) // 2) ** 2

# Example usage
n = 5
result = sum_of_cubes(n)
print("Sum of cubes of first", n, "natural numbers:", result)

Sum of cubes of first 5 natural numbers: 225


The output of this code is "Sum of cubes of first 5 natural numbers: 225" because the sum of cubes of the first 'n' natural numbers has a direct formula:

(n * (n + 1) / 2)^2

In this case, n = 5. Substituting this value into the formula, we get:

(5 * (5 + 1) / 2)^2 = (5 * 6 / 2)^2 = 15^2 = 225

Therefore, the sum of cubes of the first 5 natural numbers is 225.

